python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -56,7 +58,7 @@ class TemporalContext(NamedTuple):
     last_event_id: Optional[int]
 
 
-class AssetGraphView:
+class AssetGraphView(LoadingContext):
```

Comment:
my initial reaction is a bit weary about having several `LoadingContext` s 

I feel like ideally we we would use the `WorkspaceRequestContext` for caching within the scope of a web request or daemon task

Reply:
My thinking here is that it's sortof a pain to create a WorkspaceRequestContext in a lot of scenarios (particularly unit tests, but also we don't feed through the WorkspaceRequestContext in a way that's accessible to user-space sensors, which AutomationConditions will eventually sometimes be evaluated in).

To help clarify the direction a bit more, I have a plan to rename AssetGraphView to DagsterSession (or similar), and have that be a user-facing concept (basically a more constrained version of DagsterInstance with nice caching properties and knowledge of the asset graph).

This DagsterSession object is a bit different than a WorkspaceRequestContext in that it needs to know the current AssetGraph in order to function (and in GQL-land, the AssetGraph may be fairly expensive to fetch).

None of that leads to any particular conclusion, just laying out my reasoning for feeling like this is conceptually different / makes sense to be independent of the WorkspaceRequestContext.

Reply:
> accessible to user-space

I find this compelling 

I still worry about the types of problems that can arise if a request context and session are both in scope in like a daemon and things are being cached/cleared independently in a way thats tricky to work through. Maybe the session should proxy back to the request context when it makes sense.